osu.Game/Users/UserPanel.cs

```diff
@@ -227,8 +232,16 @@ private void displayStatus(UserStatus status)
                 statusBar.FadeIn(transition_duration, Easing.OutQuint);
                 this.ResizeHeightTo(height, transition_duration, Easing.OutQuint);
 
-                statusMessage.Text = status.Message;
+                if (status is UserStatusOnline && activity != null)
+                {
+                    statusMessage.Text = activity.Status;
+                    statusBg.FadeColour(activity.GetAppropriateColour(colours), 500, Easing.OutQuint);
+                    return;
```

Comment:
this early return is awkward. i'd move this out of the parent `else` and combine with the two lines outside into a second if-else.